### PR TITLE
Update the apollo cache

### LINKED ISSUE #1391: 🐛 [Bug] Deleted Comment reappears after editing another
> [datenbrei](https://github.com/datenbrei) **Authored by [datenbrei](https://github.com/datenbrei)**
_<time datetime="2019-08-26T09:15:40Z" title="Monday, August 26th 2019, 11:15:40 am +02:00">Aug 26, 2019</time>_
_Closed <time datetime="2019-08-28T17:38:00Z" title="Wednesday, August 28th 2019, 7:38:00 pm +02:00">Aug 28, 2019</time>_
---

## :bug: Bugreport

A once deleted comment reappears, when another comment on the same post ist edited.

### Steps to reproduce the behavior
1. Write 4 Comments
Bildschirmfoto_2019-08-26_11-03-18

2. Delete number 2
Bildschirmfoto_2019-08-26_11-03-52

3. Edit number 3 and save
4. Number 2 appears again
Bildschirmfoto_2019-08-26_11-04-19

### Expected behavior

A deleted comment should be gone for one and ever.

### Version & Environment
 Type: [Desktop]      
 - OS: [Linux]      
 - Browser: [Firefox] 
 - Version [68.0.1]  
 - Device: [Notebook]  

### Additional context

This does not happen, when the whole page is reloaded after deleting post number 2.

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-26T16:56:22Z" title="Monday, August 26th 2019, 6:56:22 pm +02:00">Aug 26, 2019</time>_
_Merged <time datetime="2019-08-28T17:38:00Z" title="Wednesday, August 28th 2019, 7:38:00 pm +02:00">Aug 28, 2019</time>_
---

### Issues

- fixes #1391

### CODE REVIEW

File: webapp/components/Comment.vue [hunk 1]
```diff
@@ -142,16 +143,23 @@ export default {
...
         await this.$apollo.mutate({
-          mutation: gqlMutation,
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: async store => {
+            const data = await store.readQuery({
+              query: PostQuery(this.$i18n),
+              variables: { id: this.post.id },
+            })
+
```
Comment (Tirokk) [inline webapp/components/Comment.vue:154]: > [appinteractive](https://github.com/appinteractive) **Authored by [appinteractive](https://github.com/appinteractive)**
_<time datetime="2019-08-27T07:18:26Z" title="Tuesday, August 27th 2019, 9:18:26 am +02:00">Aug 27, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -140,24 +141,33 @@ export default {
       this.openEditCommentMenu = showMenu
       this.setEditPending(showMenu)
     },
-    async deleteCommentCallback() {
-      try {
-        var gqlMutation = gql`
-          mutation($id: ID!) {
-            DeleteComment(id: $id) {
-              id
-            }
-          }
-        `
-        await this.$apollo.mutate({
-          mutation: gqlMutation,
+    deleteCommentCallback() {
+      this.$apollo
+        .mutate({
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: store => {
```


`update: async store => {`

File: webapp/components/Comment.vue [hunk 2]
```diff
@@ -142,16 +143,23 @@ export default {
...
+            const index = data.Post[0].comments.findIndex(
```
Comment (Tirokk) [inline webapp/components/Comment.vue:155]: > [appinteractive](https://github.com/appinteractive) **Authored by [appinteractive](https://github.com/appinteractive)**
_<time datetime="2019-08-27T07:19:50Z" title="Tuesday, August 27th 2019, 9:19:50 am +02:00">Aug 27, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -140,24 +141,33 @@ export default {
       this.openEditCommentMenu = showMenu
       this.setEditPending(showMenu)
     },
-    async deleteCommentCallback() {
-      try {
-        var gqlMutation = gql`
-          mutation($id: ID!) {
-            DeleteComment(id: $id) {
-              id
-            }
-          }
-        `
-        await this.$apollo.mutate({
-          mutation: gqlMutation,
+    deleteCommentCallback() {
+      this.$apollo
+        .mutate({
+          mutation: CommentMutations(this.$i18n).DeleteComment,
           variables: { id: this.comment.id },
+          update: store => {
+            const data = store.readQuery({
```


`const data = await store.readQuery(`

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-27T07:59:39Z" title="Tuesday, August 27th 2019, 9:59:39 am +02:00">Aug 27, 2019</time>_

---


hey thanks @appinteractive... the issue was that I was trying to put the update outside of the closing `}`, but inside the closing `)`... when I moved it inside both, it worked fine. I added the `async/awaits` to the update, readQuery, and writeQuery, as you suggested. I have a couple questions, if/when you have time to answer them.
1) is there any benefit to using `try/catch(async/await)` over `promises/then` syntax? I feel like it's just a style thing, but would love to know if there are any benefits/downsides to each.
2) the docs don't use `async/await` for updates or `readQuery`/`writeQuery`, is that cause the examples are in smart queries, or some other reason you know of?

**Review by Mogge — APPROVED** [review]
Looks good to me, just stumbled over #1414 but this is also present on `master`.

approved

**Pull request closed without merging**